### PR TITLE
Adds convex hull analysis

### LINKED ISSUE #8883: Add Convex Hull Analysis

### LINKED ISSUE #9326: Write on boarding text for "Convex Hull" analysis
We are about to add the "Convex Hull" analysis (https://github.com/CartoDB/cartodb/issues/8883) and would need some information for the screen that appears after an analysis is finished.

### PR DESCRIPTION
**Pull request by javierarce**
This PR adds the group-points family of analyses: `concave-hull`, `convex-hull`, `bounding-box`, and `bounding-circle`.

screen shot 2016-08-19 at 11 44 21
---
screen shot 2016-08-19 at 11 53 15
---
screen shot 2016-08-19 at 11 53 32

Waiting for https://github.com/CartoDB/camshaft/pull/175 to merge it.

Closes #8883 
Closes #9326

### CODE REVIEW

**Comment by javierarce** [comment]
@xavijam, can you review the code, please? 

The build is failing because some analyses aren't in the reference yet: 

`Error: no analysis params found for type: bounding-box`

**Comment by xavijam** [comment]
Hey, some screenshots could help :)

**Comment by javierarce** [comment]
Sorry! I'll add some.

**Comment by javierarce** [comment]
I added some images.

Camshaft references for the analyses are here:
[bounding-circle](https://github.com/CartoDB/camshaft/pull/175/files#diff-4cfd85cf455a2b2c4358f337d8e864dbR64)
[bounding-box](https://github.com/CartoDB/camshaft/pull/175/files#diff-4cfd85cf455a2b2c4358f337d8e864dbR34)
[convex-hull](https://github.com/CartoDB/camshaft/pull/175/files#diff-4cfd85cf455a2b2c4358f337d8e864dbR194)
[concave-hull](https://github.com/CartoDB/camshaft/pull/175/files#diff-4cfd85cf455a2b2c4358f337d8e864dbR154)

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points-form-model.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      target_percent: {
+        type: 'Number',
+        label: 'Target percent',
+        validators: ['required']
+      },
+      allow_holes: {
+        type: 'Radio',
+        text: 'Allow holes',
+        options: [
+          { val: false, label: 'No' },
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points-form-model.js]: en.json?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points/bounding-box.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    };
+
+    return {
+      type: 'bounding-box',
+      aggregate: aggregate
+    };
+  },
+
+  formatAttrs: function (formAttrs) {
+    delete formAttrs.aggregate;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points/bounding-box.js]: Be careful with delete references in an object, if it is being used in other place. If so, consider use `_.omit`, for example. Reference: https://github.com/CartoDB/cartodb/issues/9293

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points/bounding-circle.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      attribute: nodeAttrs.aggregation === 'count' ? '' : nodeAttrs.aggregation_column
+    };
+    return {
+      type: 'bounding-box',
+      aggregate: aggregate
+    };
+  },
+
+  formatAttrs: function (formAttrs) {
+    delete formAttrs.aggregate;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points/bounding-circle.js]: Same here.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points/concave-hull.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      attribute: nodeAttrs.aggregation === 'count' ? '' : nodeAttrs.aggregation_column
+    };
+    return {
+      type: 'concave-hull',
+      aggregate: aggregate
+    };
+  },
+
+  formatAttrs: function (formAttrs) {
+    delete formAttrs.aggregate;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points/concave-hull.js]: Dixit!

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points/convex-hull.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      attribute: nodeAttrs.aggregation === 'count' ? '' : nodeAttrs.aggregation_column
+    };
+    return {
+      type: 'convex-hull',
+      aggregate: aggregate
+    };
+  },
+
+  formatAttrs: function (formAttrs) {
+    delete formAttrs.aggregate;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/group-points/convex-hull.js]: 🎹

**Comment by xavijam** [comment]
For me code looks 👍, just a comment about "delete"/omit references. But I feel we should add some tests in order to check analysis result is ok depending of the type chosen.

**Comment by ohasselblad** [comment]
Small thing about text formatting: All other analyses capitalize first letter only. This one should be changed to "Group points into polygons" instead

**Comment by javierarce** [comment]
screen shot 2016-08-24 at 11 12 09

**Comment by xavijam** [comment]
Looks pretty nice!! \o/, great job @javierarce.
BTW, what about those tests I mentioned before :P?

**Comment by javierarce** [comment]
Oh, I'll add them.

**Comment by xavijam** [comment]
:clap: thanks man! :)

**Comment by xavijam** [comment]
On premise frontend tests failed, any clue @jvillarf?

**Comment by xavijam** [comment]
Problem:

16:37:43 Warning: PhantomJS timed out, possibly due to an unfinished async spec. Use --force to continue.

I feel we are safe to merge @javierarce.

**Comment by AbelVM** [comment]
Looks like we are applying minimum bounding circle to 4326, that's the reason of the elliptic result. The analysis should translate first to 3857, calc the circle, and then back to 4326.

**Comment by javierarce** [comment]
@AbelVM: could you open a ticket in camshaft, please? cc: @dgaubert

**Comment by rochoa** [comment]
@AbelVM what is correct in terms of data/analysis?

**Comment by AbelVM** [comment]
https://github.com/CartoDB/camshaft/issues/183

**Comment by rochoa** [comment]
It would be interesting if you explain why it should be that way.

**Comment by AbelVM** [comment]
Sure. The issue here is related to the haversine formula, and the deformation is the same as when you try to make a square 1ºx1º... you will get a rectangle instead. The further from the Equator you go, the elonged the ellipse will be. Spherical coordinates in a cylindric world.

From my point of view, if the user asks for a minimum bounding circle, we should return a "real life" circle.

**Pull request merged**